docs/guide/index.md

```diff
@@ -17,16 +17,149 @@ export default defineConfig({
+    plugins: [
+        DynamicPublicDirectory( dirAssets, {
+            ssr: true,
+            mimeTypes
+        }) as PluginOption
+    ],
+    publicDir: false,
+})
+```
+* With the above configuration will automatically add files in `libs/assets`, `repo1/assets` folders as static assets for your project
+* It also recommended to set `publicDir` to `false` to avoid confusion
+* Notice the wildcard `**`, this plugin use glob pattern by default. You could also review [fast-glob](https://www.npmjs.com/package/fast-glob) and [micromatch](https://www.npmjs.com/package/micromatch)
+
+### Astro Support
 
-- dirAssets: list of your static directories
-- mimeTypes: Add your extended content types, by default there will be the following content types
-- ssr: default is `false` support client side, using with Solid-js or framework support SSR please enable this flag to `true`
+In `astro.config.mjs`
+```ts
+import {defineConfig} from 'astro/config';
+import {astroMultipleAssets} from "vite-multiple-assets";
+
+const assets = ['public2/**'];
+// https://astro.build/config
+export default defineConfig({
+    integrations: [
+        astroMultipleAssets(assets)
+    ],
+});
+```
+
+### Example
+[Detail](https://github.com/nguyenbatranvan/vite-multiple-assets/blob/main/packages/examples/react/vite.config.ts)
+
+## Options
+
+```ts
+import type { PluginOption } from "vite";
+
+export default function DynamicPublicDirectory(assets: IAssets, opts: IConfig = {}): PluginOption
+```
 
 ```ts
+import type { Options } from "fast-glob";
+
+export interface IConfigExtend extends Partial<Pick<Options, "ignore" | "dot">> {
+    dst?: string | FDst;
+}
+
+export interface IConfig extends 
+    IConfigExtend, 
+    Partial<Pick<Options, "onlyFiles" | "onlyDirectories" | "cwd" | "markDirectories">> 
 {
+    __dst?: string; // NOTE: internal destination from parsing rollup write bundlers nor vite config.
+    mimeTypes?: IMIME;
+    ssr?: boolean;
+}
+```
+
+<!-- NOTE: for documenters, these options is ordered from the most important to the less -->
+<!-- NOTE: types preview using `var` instead of `let` and `const` as function arguments are `var` by design -->
+
+### `assets`
+
+```ts
+var assets: IAssets = [];
+type IAssets = string[];
+```
+
+List copied `assets` by following pattern defined in [`fast-glob` pattern](https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#pattern-syntax) which is [Unix Glob](https://man7.org/linux/man-pages/man7/glob.7.html) [Pattern](https://www.gnu.org/software/bash/manual/html_node/Pattern-Matching.html). There are few things to take:
+
+- always use forward-slash `/` for path, and use backslash to escape character
+- do not use single-dot `./**/*.txt` `./abc/**/*.txt` for current path, instead straight use the pattern `**/*.txt` `abc/**/*.txt`
+- If you are hope certain folder included to be copied, make it in selectable list pattern, `dir1/{\x01,folder_name}/**`. Just ensure there is no folder or file named by first ascii
```

Comment:
I don't think so. But, I believe this section could works?

Reply:
only `dir1/folder_name/**` will ignore `folder_name`, i tried it and it worked

Reply:
> I don't think so. But, I believe this section could works?

Oh. I mean by "works" as "works to be a guidance for user".